web/src/api/getState.ts

```diff
@@ -21,30 +32,55 @@ const getState = async (timeAverage: string): Promise<GridState> => {
 };
 
 const useGetState = (): UseQueryResult<GridState> => {
-  const [timeAverage] = useAtom(timeAverageAtom);
-  const isHourly = useAtomValue(isHourlyAtom);
+  const queryClient = useQueryClient();
+  const { urlTimeAverage, urlDatetime } = useParams<{
+    urlTimeAverage: UrlTimeAverages;
+    urlDatetime?: string;
+  }>();
+  const isHourly = urlTimeAverage
+    ? URL_TO_TIME_AVERAGE[urlTimeAverage] === TimeAverages.HOURLY
+    : false;
 
-  // First fetch last hour only
-  const last_hour = useQuery<GridState>({
+  const shouldUseLastHour = isHourly && !urlDatetime;
+  const timeAverage = urlTimeAverage
+    ? URL_TO_TIME_AVERAGE[urlTimeAverage]
+    : TimeAverages.HOURLY;
+  console.log('shouldUseLastHour', shouldUseLastHour, isHourly, urlTimeAverage);
+  const lastHourQuery = useQuery<GridState>({
     queryKey: [QUERY_KEYS.STATE, { aggregate: 'last_hour' }],
-    queryFn: async () => getState('last_hour'),
-    enabled: isHourly,
+    queryFn: () => getState('last_hour'),
+    enabled: shouldUseLastHour,
   });
 
-  const hourZeroWasSuccessful = Boolean(last_hour.isLoading === false && last_hour.data);
+  const fullDataQuery = useQuery<GridState>({
+    queryKey: [
+      QUERY_KEYS.STATE,
+      {
+        aggregate: timeAverage,
+        targetDatetime: urlDatetime,
+      },
+    ],
+    queryFn: () => getState(timeAverage, urlDatetime),
+    enabled: !shouldUseLastHour || (shouldUseLastHour && lastHourQuery.isSuccess),
+  });
 
-  const shouldFetchFullState =
-    !isHourly || hourZeroWasSuccessful || last_hour.isError === true;
+  useEffect(() => {
+    if (shouldUseLastHour && fullDataQuery.data) {
+      queryClient.setQueryData(
+        [QUERY_KEYS.STATE, { aggregate: 'last_hour' }],
+        fullDataQuery.data
+      );
+    }
+  }, [shouldUseLastHour, fullDataQuery.data, queryClient]);
```

Comment:
Should we just remove the usage of last_hour?

It seems to add a lot of complexity for very small gains now that we have the cloudflare cache in place.

Reply:
I think it's working okay here. If we want to remove it let's take that in a separate PR